### PR TITLE
Add new sample: azure-spring-boot-sample-keyvault-certificates-client-side.

### PR DESCRIPTION
**Pull request by chenrujun**
Add new sample: azure-spring-boot-sample-keyvault-certificates-client-side.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @saragluna , @yiliuTo .
Please review this PR.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Azure Spring Boot Key Vault Certificates client library for Java
+
+## Key concepts
+This sample illustrates how to use [Azure Spring Boot Starter Key Vault Certificates ][azure_spring_boot_starter_key_vault_certificates].
+
+This sample should work together with [azure-spring-boot-sample-keyvault-certificates-server-side].
+
+## Getting started
+
+### Prerequisites
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md:10]: Let's use unified section captions and structures of all sample projects.
Prerequisites -> Environment checklist

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- [Environment checklist][environment_checklist]
+
+### Run this application
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md:13]: Move the whole section as a sub under "Examples" and remove this caption.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+#### Add configurations
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md:15]: #### Add configurations -> ### Config the sample

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Fulfill these properties in application.yml:
+```yaml
+azure:
+  keyvault:
+    uri:                 # The URI to the Azure Key Vault used
+    tenant-id:           # The Tenant ID for your Azure Key Vault (needed if you are not using managed identity).
+    client-id:           # The Client ID that has been setup with access to your Azure Key Vault (needed if you are not using managed identity).
+    client-secret:       # The Client Secret that will be used for accessing your Azure Key Vault (needed if you are not using managed identity).
+```
+
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md:25]: Add a section caption "### How to run" before below things.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Review by michaelqi793 — COMMENTED** [review]
LGTM!

**Comment by chenrujun** [comment]
Hi, @saragluna , @yiliuTo .
Please help to review this PR when you have time.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+### How to run
+1. Start azure-spring-boot-sample-keyvault-certificates-server-side's SampleApplication
+1. Start azure-spring-boot-sample-keyvault-certificates-client-side SampleApplication
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md:26]: inconsistent between azure-spring-boot-sample-keyvault-certificates-server-side'**s** and azure-spring-boot-sample-keyvault-certificates-client-side

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/pom.xml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    </dependency>
+
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-web</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-actuator</artifactId>
+    </dependency>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/pom.xml:43]: Is this a required scope?

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-server-side/README.md [hunk 7]
```diff
@@ -13,12 +15,32 @@ In this sample, a certificate named `self-signed` is stored into an Azure Key Va
...
+azure:
+  keyvault:
+    uri:                 # The URI to the Azure Key Vault used
+    tenant-id:           # The Tenant ID for your Azure Key Vault (needed if you are not using managed identity).
+    client-id:           # The Client ID that has been setup with access to your Azure Key Vault (needed if you are not using managed identity).
+    client-secret:       # The Client Secret that will be used for accessing your Azure Key Vault (needed if you are not using managed identity).
+```
+
+### How to run
+1. SampleApplication
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-server-side/README.md:30]: How about adding a verb here.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md [hunk 8]
```diff
@@ -100,27 +100,24 @@ This starter allows you to securely manage and tightly control your certificates
 
     ![Copy secrets](resource/copy-secrets.png)
 
-To use the starter for server side SSL, you will need to add the following to
-your `application.yml` (if the application is using Spring Cloud Config 
-Server for its configuration add it to the `bootstrap.yml` of the application)
-
+Add these items in your `application.yml`:
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md:103]: Not this line but the above `Using a client id and client secret` section. 

We already have this doc https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot-samples/create-sp-using-azure-cli.md, and I think the steps of creating a service principal via Azure Portal is common to most of our starters. How about we put these steps and screenshots in the same place?

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md [hunk 9]
```diff
@@ -157,148 +152,139 @@ server:
...
-    return restTemplate;
+    return new RestTemplate(requestFactory);
 }
 ```
 
-### Enable mutual SSL on the client side
 
-Only some minor changes need to be done to the client side SSL example 
-mentioned above.
+### Enable mutual SSL (MTLS).
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md:238]: mTLS

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java [hunk 10]
```diff
@@ -3,112 +3,105 @@
...
+import javax.net.ssl.HttpsURLConnection;
+import java.security.Security;
+import java.util.Optional;
+import java.util.Properties;
 
 /**
  * Leverage {@link EnvironmentPostProcessor} to add Key Store property source.
  */
-@Order(LOWEST_PRECEDENCE)
+@Order
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java:24]: why are we removing the LOWEST_PRECEDENCE.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/pom.xml (hunk 6 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/pom.xml:43]: No, it's used for debug. The server-side sample also have this dependency. I think it's OK to keep it.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md (hunk 8 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md:103]: Issue created: https://github.com/Azure/azure-sdk-for-java/issues/20143
I'll do that in another PR.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java (hunk 10 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java:24]: `LOWEST_PRECEDENCE` is the default value. I just accept Intellij's suggestion.

**Review by saragluna — APPROVED** [review]

**Pull request merged**